benchopt/cli/main.py

```diff
@@ -182,15 +188,21 @@ def run(config_file=None, **kwargs):
 
     (
         benchmark, solver_names, forced_solvers, dataset_names,
-        objective_filters, max_runs, n_repetitions, timeout, n_jobs, slurm,
+        objective_filters, max_runs, n_repetitions, timeout, no_timeout, n_jobs, slurm,
         collect, plot, display, html, pdb, do_profile, env_name, output_name
     ) = _get_run_args(kwargs, config)
 
-    try:
-        timeout = int(float(timeout))
-    except ValueError:  # already under string format
-        import pandas as pd
-        timeout = pd.to_timedelta(timeout).total_seconds()
+    # If --no-timeout is set and --timeout is not, skip this block and keep timeout=None
+    if timeout != None and no_timeout:
```

Comment:
To check if a variable is `None` or not, it is usually better to use `is/is not`
```suggestion
    if timeout is not None and no_timeout:
```